lib/build/utils.js

```diff
@@ -138,3 +131,89 @@ function modifyFilename(pth, modifier) {
   let ext = path.extname(pth);
   return path.posix.join(path.dirname(pth), modifier(path.basename(pth, ext), ext));
 }
+
+// https://nodejs.org/dist/latest-v10.x/docs/api/modules.html
+// after "high-level algorithm in pseudocode of what require.resolve() does"
+function nodejsLoadAsFile(resourcePath) {
+  if (fs.isFile(resourcePath)) {
+    return resourcePath;
+  }
+
+  if (fs.isFile(resourcePath + '.js')) {
+    return resourcePath + '.js';
+  }
+
+  if (fs.isFile(resourcePath + '.json')) {
+    return resourcePath + '.json';
+  }
+  // skip .node file that nobody uses
+}
+
+function nodejsLoadIndex(resourcePath) {
+  if (!fs.isDirectory(resourcePath)) return;
+
+  const indexJs = path.join(resourcePath, 'index.js');
+  if (fs.isFile(indexJs)) {
+    return indexJs;
+  }
+
+  const indexJson = path.join(resourcePath, 'index.json');
+  if (fs.isFile(indexJson)) {
+    return indexJson;
+  }
+  // skip index.node file that nobody uses
+}
+
+function nodejsLoadAsDirectory(resourcePath) {
+  if (!fs.isDirectory(resourcePath)) return;
+
+  const packageJson = path.join(resourcePath, 'package.json');
+
+  if (fs.isFile(packageJson)) {
+    let metadata;
+    try {
+      metadata = JSON.parse(fs.readFileSync(packageJson));
+    } catch (err) {
+      console.error(err);
+      return;
+    }
+    let metaMain;
+    // try 1.browser > 2.module > 3.main
+    // the order is to target browser.
+    // when aurelia-cli introduces multi-targets build,
+    // it probably should use different order for electron app
+    // for electron 1.module > 2.browser > 3.main
+    if (typeof metadata.browser === 'string')  {
+      // use package.json browser field if possible.
+      metaMain = metadata.browser;
+    } else if (typeof metadata.module === 'string' &&
+      !(metadata.name && metadata.name.startsWith('aurelia-'))) {
+      // prefer es module format over cjs, just like webpack.
+      // this improves compatibility with TypeScript.
+      // ignores aurelia-* core npm packages as their module
+      // field is pointing to es2015 folder.
+      metaMain = metadata.module;
+    } else if (typeof metadata.main === 'string') {
+      metaMain = metadata.main;
+    }
+
+    let mainFile = metaMain || 'index';
+    const mainResourcePath = path.resolve(resourcePath, mainFile);
+    return nodejsLoadAsFile(mainResourcePath) || nodejsLoadIndex(mainResourcePath);
+  }
+
+  return nodejsLoadIndex(resourcePath);
+}
+
+exports.nodejsLoad = function(resourcePath) {
+  return nodejsLoadAsFile(resourcePath) || nodejsLoadAsDirectory(resourcePath);
+};
+
+exports.removeJsExtension = function(filePath) {
+  if (path.extname(filePath).toLowerCase() === '.js') {
+    return filePath.substr(0, filePath.length - 3);
```

Comment:
We may want to use `slice(0, ... -3)` instead of this, according to this https://github.com/jsdom/jsdom/pull/2419

Though it's negligible, still feel good.